std/file.d

```diff
@@ -3200,6 +3200,219 @@ version(Posix) unittest //issue 11434
     assert(getAttributes(t2) == octal!100767);
 }
 
+/***************************************************
+Copy file or directory $(D from) to $(D to). When a directory is copied, its
+contents are copied as well, recursively.
+
+File timestamps are preserved, except for symbolic links.
+File attributes are preserved, if $(D preserve) equals $(D PreserveAttributes.yes).
+On Windows only $(D PreserveAttributes.yes) (the default on Windows) is supported.
+
+Existing target files are overwritten. Existing target directories are written
+into.
+
+With symbolic links, the links themselves are copied. That is, for every
+symbolic link in $(D from), a new link with the same target path is created in
+$(D to). Relative target paths are not changed. They may be invalidated by the
+copying, when they point outside of the copied directory tree.
+
+Special files (e.g. device files, FIFOs) cannot be copied with $(D copyRecurse).
+
+Params:
+    from = string or range of characters representing the existing
+        file/directory name
+    to = string or range of characters representing the target file/directory
+        name
+    preserve = whether to preserve the file attributes
+
+Throws: $(D FileException) on error.
+ */
+void copyRecurse(string from, string to,
+    PreserveAttributes preserve = preserveAttributesDefault)
+{
+    // Remove any trailing directory separators.
+    from = buildNormalizedPath(from);
+    to = buildNormalizedPath(to);
+
+    immutable attrs = getLinkAttributes(from);
+
+    if (attrIsFile(attrs)) copy(from, to, preserve);
```

Comment:
I'd be perfectly fine with a single function for removing, and I'm fine with multiple functions given that one of them does remove everything. Similarly, if we want to have multiple copy functions, just so long as one of them works like cp and will take a target directory, I'm fine with it, much as I think that the other is too annoying to be useful. However, I don't know why `copyRecurse` would require a file as the target rather than accepting a directory. But then again, I'd expect `copyRecurse` to function like `cp` does, and looking over the unit tests, it doesn't look like it does.

What I want is basically `cp` but without having to call to the shell and deal with escaping file names or any other quirks of the shell. If you guys want to have other functions that are much pickier, then fine, but personally, I have no interest in ever using them and think that they're a waste of space.